PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetExtractor.cxx

```diff
@@ -630,7 +632,7 @@ void AliAnalysisTaskJetExtractor::UserCreateOutputObjects()
 
   // ### Initialize the jet tree (settings must all be given at this stage)
   fJetTree->SetRandomGenerator(fRandomGenerator);
-  fJetTree->InitializeTree(fSaveCaloClusters, fSaveMCInformation, (fMatchedDetLevelJetsArrayName != ""), (fMatchedPartLevelJetsArrayName != ""), fSaveConstituents, fSaveConstituentsIP, fSaveConstituentPID, fSaveJetShapes, fSaveJetSplittings, fSaveSecondaryVertices, fSaveTriggerTracks);
+  fJetTree->InitializeTree(fSaveCaloClusters, fSaveMCInformation, fSaveMCInformation, fDoPartLevelMatching, fSaveConstituents, fSaveConstituentsIP, fSaveConstituentPID, fSaveJetShapes, fSaveJetSplittings, fSaveSecondaryVertices, fSaveTriggerTracks);
```

Comment:
This need to be solved differently. In the present implementation you would do the matching also if no jet containers for matching were given.
I would suggest to implement fDoPartLevelMatching and, in addition, fDoDetLevelMatching as variables which are set in case jet container 1 and jet container 2 are added to the task. Then you don't need the setter for fDoPartLevelMatching and everything is automatically set based on which containers are added. Add an information string on which matching is on/off in PrintConfig() for the bookkeeping.

If you want to keep it all explicitly activated by the user, fDoDetLevelMatching needed a setter and it would have to appear in this line.

Reply:
Yup, I was going to ask about this one, I will add fDoDetLevelMatching!